src/app/CommandHandler.h

```diff
@@ -189,32 +192,37 @@ class CommandHandler
     CommandHandler(Callback * apCallback);
 
     /*
-     * Constructor to override number of supported paths per invoke.
+     * Constructor to override the number of supported paths per invoke and command responder.
+     *
+     * The callback and any pointers passed via TestOnlyOverrides must outlive this
+     * CommandHandler object.
      *
-     * The callback and command path registry passed in has to outlive this CommandHandler object.
      * For testing purposes.
      */
-    CommandHandler(TestOnlyMarker aTestMarker, Callback * apCallback, CommandPathRegistry * apCommandPathRegistry);
+    CommandHandler(TestOnlyOverrides & aTestOverride, Callback * apCallback);
 
     /*
-     * Main entrypoint for this class to handle an invoke request.
+     * Main entrypoint for this class to handle an InvokeRequestMessage.
      *
-     * This function will always call the OnDone function above on the registered callback
-     * before returning.
+     * This function MAY call the registered OnDone callback before returning.
+     * To prevent immediate OnDone invocation, callers can wrap their CommandHandler instance
+     * within a CommandHandler::Handle.
      *
      * isTimedInvoke is true if and only if this is part of a Timed Invoke
      * transaction (i.e. was preceded by a Timed Request).  If we reach here,
      * the timer verification has already been done.
+     *
+     * commandResponder handles sending InvokeResponses, added by clusters, to the client.
      */
-    void OnInvokeCommandRequest(Messaging::ExchangeContext * ec, const PayloadHeader & payloadHeader,
-                                System::PacketBufferHandle && payload, bool isTimedInvoke);
+    Protocols::InteractionModel::Status OnInvokeCommandRequest(CommandResponderInterface & commandResponder,
```

Comment:
This preserves a pointer to commandResponder insiude via SetCommandResponder. Could we have some details about the expected lifetime of commandResponder? how long does it have to live and when is it safe to drop?